octopusdeploy_framework/datasource_lifecycles.go

```diff
@@ -0,0 +1,230 @@
+package octopusdeploy_framework
+
+import (
+	"context"
+	"fmt"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/core"
+	"github.com/OctopusDeploy/go-octopusdeploy/v2/pkg/lifecycles"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"time"
+)
+
+type lifecyclesDataSource struct {
+	*Config
+}
+
+type lifecyclesDataSourceModel struct {
+	ID          types.String `tfsdk:"id"`
+	SpaceID     types.String `tfsdk:"space_id"`
+	IDs         types.List   `tfsdk:"ids"`
+	PartialName types.String `tfsdk:"partial_name"`
+	Skip        types.Int64  `tfsdk:"skip"`
+	Take        types.Int64  `tfsdk:"take"`
+	Lifecycles  types.List   `tfsdk:"lifecycles"`
+}
+
+func NewLifecyclesDataSource() datasource.DataSource {
+	return &lifecyclesDataSource{}
+}
+
+func (l *lifecyclesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	tflog.Debug(ctx, "lifecycles datasource Metadata")
+	resp.TypeName = "octopusdeploy_lifecycles"
+}
+
+func (l *lifecyclesDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	tflog.Debug(ctx, "lifecycles datasource Schema")
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id":           schema.StringAttribute{Computed: true},
+			"space_id":     schema.StringAttribute{Optional: true},
+			"ids":          schema.ListAttribute{ElementType: types.StringType, Optional: true},
+			"partial_name": schema.StringAttribute{Optional: true},
+			"skip":         schema.Int64Attribute{Optional: true},
+			"take":         schema.Int64Attribute{Optional: true},
```

Comment:
we have helper funcs in the schema package that covers all of these attributes. It means we have consistentcy on the definitions including descriptions